motioncorr/viewers.py

```diff
@@ -24,45 +26,120 @@
-        viewParamsDef = {showj.MODE: showj.MODE_MD,
-                         showj.ORDER: labelsDef,
-                         showj.VISIBLE: labelsDef,
-                         showj.RENDER: None
-                         }
-
-        outputLabels = ['outputMicrographs', 'outputMicrographsDoseWeighted',
-                        'outputMovies']
-
-        if not any(obj.hasAttribute(l) for l in outputLabels):
-            return [self.infoMessage("Output (micrographs or movies) have "
-                                     "not been produced yet.")]
-
-        # Display only the first available output, showing all of them
-        # can be confusing and not useful.
-        # The user can still double-click in the specific output
-        for l in outputLabels:
-            if obj.hasAttribute(l):
-                output = getattr(obj, l)
-                if 'Micrographs' in l:
-                    return [MicrographsView(self.getProject(), output)]
-                else:  # Movies case
-                    return [self.objectView(output, viewParams=viewParamsDef)]
+        elif param == 'doShowMicsDW':
+            if getattr(self.protocol, 'outputMicrographsDoseWeighted', None) is not None:
+                return [MicrographsView(self.getProject(),
+                                    self.protocol.outputMicrographsDoseWeighted)]
+            else:
+                self._errors.append('No output dose-weighted micrographs found!')
 
+        elif param == 'doShowMovies':
+            if getattr(self.protocol, 'outputMovies', None) is not None:
+                labelsDef = 'enabled id _filename _samplingRate '
+                labelsDef += '_acquisition._dosePerFrame _acquisition._doseInitial '
+                viewParamsDef = {showj.MODE: showj.MODE_MD,
+                                 showj.ORDER: labelsDef,
+                                 showj.VISIBLE: labelsDef,
+                                 showj.RENDER: None
+                                 }
+                output = self.protocol.outputMovies
+                return [self.objectView(output, viewParams=viewParamsDef)]
+            else:
+                self._errors.append('No output movies found!')
+
+        elif param == 'doShowFailedMovies':
+            failedList = self.protocol._readFailedList()
+            if not failedList:
+                self._errors.append("No failed movies found!")
+            else:
+                labelsDef = 'enabled id _filename _samplingRate '
+                labelsDef += '_acquisition._dosePerFrame _acquisition._doseInitial '
+                viewParamsDef = {showj.MODE: showj.MODE_MD,
+                                 showj.ORDER: labelsDef,
+                                 showj.VISIBLE: labelsDef,
+                                 showj.RENDER: None
+                                 }
+                sqliteFn = self.protocol._getExtraPath('movies_failed.sqlite')
+                self.createFailedMoviesSqlite(failedList, sqliteFn)
```

Comment:
Should we check first if the failed sqlite already exists and not make the conversion again? Do we expect it to change...in which case it make sense to re-generate it all time.

Reply:
I expect it to change when run in streaming. Maybe we could check if protocol is finished and sqlite exists then use it as is.

Reply:
Yes, you are right. Maybe we should not create extra complications and just create it every time. I don't expect this to be run to often and not in a huge set.